Clean up "Writing consistent snapshots" section

joshuagl opened the pull request:
* Match how metadata and target filenames are referred to in the detailed
  client workflow, i.e. filename.ext -> FILENAME.EXT
* Remove suggestion to write a version of the root metadata without a
  version prefix. We only ever look for version-prefixed version of root
  metadata.

Fixes #184

mnm678 commented on tuf-spec.md:
```diff
@@ -1600,16 +1600,11 @@ function used to produce the digest because, on a read, the choice of
-because it is possible to download root metadata both with and without
-known version numbers. These steps are required because these are the only
-metadata files that may be requested without known version numbers.
+Timestamp metadata (timestamp.EXT) SHOULD be written to non-volatile storage
```
per RFC 2119, I think this should be a MUST

joshuagl replied:
I will make that change. I was thinking of SHOULD in the context of #185, but I should handle that separately.

mnm678 commented on tuf-spec.md:
```diff
@@ -1586,12 +1586,12 @@ We now explain how a repository should write metadata and targets to
 
 On the other hand, consistent target files should be written to
-non-volatile storage as digest.filename.ext.  This means that if the
+non-volatile storage as HASH.FILENAME.EXT.  This means that if the
```
In general I like `DIGEST` better than `HASH`, but I know that we use `HASH` elsewhere in the spec. No need to change it unless others feel strongly.

joshuagl replied:
I used `HASH` because that's what we use in the the [5.7.3](https://theupdateframework.github.io/specification/v1.0.25/#fetch-target) of the detailed client workflow and the name of the attribute in metadata is 'hashes'.

mnm678 commented on tuf-spec.md:
```diff
@@ -1586,12 +1586,12 @@ We now explain how a repository should write metadata and targets to
 produce self-contained consistent snapshots.
 
 Simply put, TUF should write every metadata file as such: if the
```
I think this and the below paragraph should also be MUSTs.

joshuagl replied:
I've added quite a few more RFC 2119 keywords in 6c54aa8 and cleaned up other paragraphs in the section.

joshuagl commented:
Thanks for the review @mnm678, I've addressed your comments and clarified some more of the text in the "Writing consistent snapshots" section. Please take a look.

joshuagl commented:
@lukpueh @trishankatdatadog could we get a second review here? Thank you!

lukpueh commented on tuf-spec.md:
```diff
@@ -1585,49 +1585,44 @@ without interrupting that client.
+Timestamp metadata (timestamp.EXT) MUST be written to non-volatile storage
+without a version prefix whenever it is updated. This is required because
+timestamp metadata is the only metadata file that may be requested without known
+version numbers.  It is OPTIONAL for an implementation to write identical copies
+at VERSION_NUMBER.timestamp.EXT for record-keeping purposes.
```
Just took me a while to parse the sentence. For me it would have been helpful to read e.g. like so

```suggestion
version numbers.  It is OPTIONAL for an implementation to write identical copies of timestamp.EXT to the respective VERSION_NUMBER.timestamp.EXT for record-keeping purposes.
```

lukpueh reviewed (approved):
Great set of clarifications, @joshuagl! I added one minor suggestion. Feel free to ignore if you think the phrase is obvious without it.

joshuagl commented:
Thanks for the review both. I committed Lukas' change, which dismissed existing reviews. Please take another look @mnm678 @lukpueh

trishankatdatadog commented on tuf-spec.md:
```diff
@@ -1585,49 +1585,45 @@ without interrupting that client.
+Timestamp metadata (timestamp.EXT) MUST be written to non-volatile storage
+without a version prefix whenever it is updated. This is required because
+timestamp metadata is the only metadata file that may be requested without known
+version numbers.  It is OPTIONAL for an implementation to write identical copies
```
```suggestion
version numbers.  It is OPTIONAL for an implementation to write an identical copy
```

joshuagl replied:
💯 thanks

joshuagl commented:
Thanks for the suggestion Trishank. I've committed it and dismissed all reviews. Please take another look all @mnm678 @lukpueh @trishankatdatadog

lukpueh reviewed (approved):
Still LGTM!
